src/lib/logical_query_plan/join_node.hpp

```diff
@@ -83,6 +96,19 @@ class JoinNode : public EnableMakeForLQPNode<JoinNode>, public AbstractLQPNode {
   bool _is_semi_reduction = false;
   mutable std::weak_ptr<JoinNode> _reduced_join_node;
 
+  /**
+   * The following flag is set by the ColumnPruningRule. It indicates whether and which input side
+   * of a join is no longer needed in the LQP tree above the join and could therefore be removed
+   * during optimization by other rules such as the JoinToSemiJoinRule and the JoinToPredicateRewriteRule.
+   * 
+   * Example: SELECT c_name FROM customer, nation WHERE c_nationkey = n_nationkey AND n_name = 'GERMANY'
+   * In this target query, the table nation is purely used for filtering out tuples from the customer table. All
+   * attributes of the nation table are completely projected away afterwards.
+   * 
+   * nullopt if both sides are used further up in the LQP
+   */
+  std::optional<LQPInputSide> _removable_input_side = std::nullopt;
```

Comment:
Here and above: since the actual rule and termini within the optimizer is about "column pruning", I wonder if we should better `_prunable_input_side`.